ape_solidity/compiler.py

```diff
@@ -184,58 +182,56 @@ def compile(
             else:
                 solc_version = max(self.installed_versions)
 
-        if not base_path:
-            # Does support base_path
-            from ape import config
+            if solc_version not in files_by_solc_version:
+                files_by_solc_version[solc_version] = set()
 
-            base_path = config.contracts_folder
+            files_by_solc_version[solc_version].add(path)
 
```

Comment:
Add an assertion that we didn't lose any files by doing this (e.g. `set(contract_filepaths)` is the same as the aggregated set of all files by version)

Reply:
I would rather cover this in tests!

Reply:
This is already covered in the tests really.